### PR TITLE
refactor: remove rc-util/lib/Dom/addEventListener

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactor: remove rc-util/lib/Dom/addEventListener |
| 🇨🇳 Chinese | 移除 addEventListener 方法，用原生代替 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a8230ba</samp>

Improved the Chinese translation of the `Affix` component documentation in `components/affix/index.zh-CN.md`. Replaced the ambiguous word "固定" with the more specific "固钉".

### <samp>🤖 Generated by Copilot at a8230ba</samp>

* Replace the term "固定" with "固钉" in the Chinese documentation of the `Affix` component to match the component name and the English documentation ([link](https://github.com/ant-design/ant-design/pull/42464/files?diff=unified&w=0#diff-58d7ecaf1d80315eef055e5f5390b4268c1d65ab84765db1983a3ca025b15028L26-R26),[link](https://github.com/ant-design/ant-design/pull/42464/files?diff=unified&w=0#diff-58d7ecaf1d80315eef055e5f5390b4268c1d65ab84765db1983a3ca025b15028L37-R37))

### CODE REVIEW

File: components/affix/index.zh-CN.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ group:
 | offsetBottom | 距离窗口底部达到指定偏移量后触发 | number | - |
 | offsetTop | 距离窗口顶部达到指定偏移量后触发 | number | 0 |
 | target | 设置 `Affix` 需要监听其滚动事件的元素，值为一个返回对应 DOM 元素的函数 | () => HTMLElement | () => window |
-| onChange | 固定状态改变时触发的回调函数 | (affixed?: boolean) => void | - |
+| onChange | 固钉状态改变时触发的回调函数 | (affixed?: boolean) => void | - |
```
Comment (afc163) [inline components/affix/index.zh-CN.md:37]: 固定是对的。

Comment (afc163) [reply components/affix/index.zh-CN.md:37]: 语义上其实是 `（固钉的）固定状态`

Comment (li-jia-nan) [reply components/affix/index.zh-CN.md:37]: 我还以为是写错了，关了

**Comment by li-jia-nan** [comment]
去年把其它地方的 addEventListener 都删掉了，遗留了这里

File: components/affix/__tests__/Affix.test.tsx [hunk 2]
```diff
@@ -157,23 +154,6 @@ describe('Affix Render', () => {
...
 
-    it('instance change', async () => {
-      const container = document.createElement('div');
-      document.body.appendChild(container);
-      let target: HTMLDivElement | null = container;
-
-      const getTarget = () => target;
-      const { rerender } = render(<Affix target={getTarget}>{null}</Affix>);
-      await waitFakeTimer();
-      expect(getObserverEntities()).toHaveLength(1);
```
Comment (li-jia-nan) [inline components/affix/__tests__/Affix.test.tsx:168]: `addObserveTarget` 方法干掉了，所以相关的测试用例也删掉了

**Review by afc163 — APPROVED** [review]

File: components/affix/index.tsx [hunk 3]
```diff
@@ -99,11 +105,12 @@ class InternalAffix extends React.Component<InternalAffixProps, AffixState> {
     const { prevTarget } = this.state;
     const targetFunc = this.getTargetFunc();
     const newTarget = targetFunc?.() || null;
-
     if (prevTarget !== newTarget) {
-      removeObserveTarget(this);
       if (newTarget) {
-        addObserveTarget(newTarget, this);
+        TRIGGER_EVENTS.forEach((eventName) => {
+          prevTarget?.removeEventListener(eventName, this.lazyUpdatePosition);
```
Comment (RexSkz) [inline components/affix/index.tsx:111]: 修改后，好像只有 `newTarget` 存在时才会执行 `prevTarget?.removeEventListener`？可能有监听器未移除干净的风险。

File: components/affix/index.tsx [hunk 4]
```diff
@@ -126,7 +133,17 @@ class InternalAffix extends React.Component<InternalAffixProps, AffixState> {
...
       this.timer = null;
     }
-    removeObserveTarget(this);
+    const { prevTarget } = this.state;
+    const targetFunc = this.getTargetFunc();
+    const newTarget = targetFunc?.();
+    TRIGGER_EVENTS.forEach((eventName) => {
+      newTarget?.removeEventListener(eventName, this.lazyUpdatePosition);
+    });
+    if (prevTarget !== newTarget) {
```
Comment (RexSkz) [inline components/affix/index.tsx:142]: 这里感觉有点多余了，这里 `prevTarget` 应该始终等于 `newTarget`，因为 `prevTarget` 只会在 `componentDidUpdate` 时更新，而 `componentDidUpdate` 会在每次 props、state、context 改变时执行。

我个人的理解（不敢 100% 确定，可以验证一下）：

- `componentDidMount` 时注册当前组件监听器
- `componentDidUpdate` 时销毁旧组件监听器，注册新组件监听器
- `componentWillUnmount` 时销毁当前组件监听器

Comment (li-jia-nan) [reply components/affix/index.tsx:142]: 这个判断我去掉吧

File: components/affix/index.tsx [hunk 5]
```diff
@@ -87,8 +90,11 @@ class InternalAffix extends React.Component<InternalAffixProps, AffixState> {
     if (targetFunc) {
       // [Legacy] Wait for parent component ref has its value.
       // We should use target as directly element instead of function which makes element check hard.
+      const target = targetFunc?.() || null;
```
[the remark this answers is not in the document]
Comment (li-jia-nan) [reply components/affix/index.tsx:93]: 确实有点问题，我改一下

File: components/affix/index.tsx (hunk 3 above)
Comment (li-jia-nan) [reply components/affix/index.tsx:111]: 而且这里用了可选链，确实没必要判断了，如果不存在会自己中断的

**Comment by li-jia-nan** [comment]
- @RexSkz  #42483

**Comment by afc163** [comment]
图片
不知道哪里又冒出来了。

**Comment by afc163** [comment]
找到了：https://github.com/react-component/image/blob/2f40670188b6429c6d3a6f608313be9a23eae168/src/hooks/useTouchEvent.ts#L3

**Comment by afc163** [comment]
https://github.com/react-component/image/blob/2f40670188b6429c6d3a6f608313be9a23eae168/src/getFixScaleEleTransPosition.ts#L1

rc-util/lib/Dom/css 也没必要用。

**Comment by li-jia-nan** [comment]
> 找到了：https://github.com/react-component/image/blob/2f40670188b6429c6d3a6f608313be9a23eae168/src/hooks/useTouchEvent.ts#L3

这个可以改成原生的 `addEventListener` 了

**Pull request merged**